mathjax3-ts/util/HTMLNodes.ts

```diff
@@ -0,0 +1,104 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2017 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview  Support functions for creating DOM elements
+ *
+ * @author dpvc@mathjax.org (Davide Cervone)
+ */
+
+import {OptionList} from './Options.js';
+
+/*****************************************************************/
+/*
+ *  Implements the HTMLNodes class for creting HTML elements
+ */
+
+export class HTMLNodes {
+    /*
+     * The document in which the HTML nodes will be created
+     */
+    protected _document: Document;
+
+    /*
+     * @param{Document} document  The document in which the nodes will be created
+     * @constructor
+     */
+    constructor(document: Document = null) {
+        this._document = document;
+    }
+
+    /*
+     * Set or get the document in which the elements are created
+     *
+     * @param{Document} document  The document in which to create elements
+     * @return{Document}  The document currently used for creating elements
+     */
+    public document(document: Document = null) {
```

Comment:
To be consistent with the rest of the code base this should be done with set and get.